docs/LARGE_TESTS.md

```diff
@@ -0,0 +1,165 @@
+# Large Tests: Building and Testing
+
+This guide gets you started with writing and running large tests on Snap plugins. If your plugin needs updated to the large test framework you can find step by step instructions [here](https://github.com/kjlyon/snap/blob/master/Pluginsync_for_large_tests.md)
+
+## Effective Large Tests
+
+As of 1.0 we introduced a large test framework which can be added to your plugin by running our pluginsync tool. The default large test performs the following actions:
+* use environment variable to populate docker compose specification in: `scripts/test/docker_compose.yml`
+* download the latest containers via `docker pull` and run them
+* conditionally run `scripts/test/setup.rb` before any test (use this to create test database, test service etc)
+* download and run the appropriate version of Snap per `$SNAP_VERSION`
+* scan `examples/task/*.yml` for list of tasks and metrics
+* load Snap plugins, first from the local `build/linux/x86_64/*` directory, then try `build.snap-telemetry.io` s3 bucket
+* verify plugins are loaded successfully
+* attempt to create, verify, and stop every yaml task in the examples directory.
+* shutdown and cleanup containers
+
+If this is not the appropriate behavior, you can write custom large test as `{test_name}_spec.rb` in the `scripts/test` directory.
+
+### Docker Compose
+A default `docker_compose.yml` file should be supplied by the developer and placed in `./scripts/test` directory. This will be used by the default large spec test. Additional docker compose config files can be supplied for complex test scenarios and they require their own custom_spec.rb test.
+
+Currently these environment variables are passed to the Snap container (we are investigating ways to pass additional/arbitrary ENV variable values):
```

Comment:
This sentence doesn't quite make sense to me. Are you telling someone something they have to know or something that is nice to know? That distinction could help clarify.